sdk/internal/perf/testdata/perf/sleep_perf.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/perf"
+)
+
+type sleepPerfTest struct {
+	perf.PerfTestOptions
+	secondsPerOp int
+}
+
+func (s *sleepPerfTest) GlobalSetup(ctx context.Context) error {
+	return nil
+}
+
+func (s *sleepPerfTest) GlobalCleanup(ctx context.Context) error {
+	return nil
+}
+
+func (s *sleepPerfTest) Setup(ctx context.Context) error {
+	s.secondsPerOp = int(math.Pow(2.0, float64(s.ParallelIndex)))
```

Comment:
Could this be moved to the ctor, or must it live in `Setup()`?  Is `ParallelIndex` available in the ctor?

Reply:
Yes it is, moving now.